dynamiq/storages/vector/pgvector/pgvector.py

```diff
@@ -0,0 +1,648 @@
+        create_extension: bool = True,
+        table_name: str | None = None,
+        schema_name: str | None = None,
+        dimension: int = 1536,
+        vector_function: PGVectorVectorFunction = PGVectorVectorFunction.COSINE_SIMILARITY,
+        index_method: PGVectorIndexMethod = PGVectorIndexMethod.EXACT,
+        index_name: str | None = None,
+        create_if_not_exist: bool = True,
+        content_key: str = "content",
+        embedding_key: str = "embedding",
+    ):
+        """
+        Initialize a PGVectorStore instance.
+
+        Args:
+            connection (PGVector | str): PGVector connection instance. Defaults to None.
+            client (Optional[PGVector]): PGVector client instance. Defaults to None.
+            create_extension (bool): Whether to create the vector extension (if it does not exist). Defaults to True.
+            table_name (str): Name of the table in the database. Defaults to None.
+            schema_name (str): Name of the schema in the database. Defaults to None.
+            dimension (int): Dimension of the embeddings. Defaults to 1536.
+            vector_function (PGVectorVectorFunction): The vector function to use for similarity calculations.
+                Defaults to 'cosine_similarity'.
+            index_method (PGVectorIndexMethod): The index method to use for the vector store.
+                Defaults to 'exact_nearest_neighbor_search'.
+            index_name (str): Name of the index to create. Defaults to None.
+            create_if_not_exist (bool): Whether to create the table and index if they do not exist. Defaults to True.
+            content_key (Optional[str]): The field used to store content in the storage. Defaults to 'content'.
+            embedding_key (Optional[str]): The field used to store embeddings in the storage. Defaults to 'embedding'.
+        """
+        if vector_function not in PGVectorVectorFunction:
+            raise ValueError(f"vector_function must be one of {PGVectorVectorFunction.list()}")
+        if index_method is not None and index_method not in PGVectorIndexMethod:
+            raise ValueError(f"index_method must be one of {PGVectorIndexMethod.list()}")
+
+        if isinstance(connection, str):
+            self.connection_string = connection
+            self._conn = psycopg.connect(self.connection_string)
+        elif isinstance(connection, PGVector):
+            self._conn = connection.connect()
+            self.connection_string = connection.conn_params
+        else:
+            raise ValueError("connection must be a string or PGVector object")
+
+        self.create_extension = create_extension
+        if self.create_extension:
+            self._conn.execute("CREATE EXTENSION IF NOT EXISTS vector")
+            self._conn.commit()
+
+        register_vector(self._conn)
+
+        self.client = client if client else self._conn
+
+        self.table_name = table_name if table_name else DEFAULT_TABLE_NAME
```

Comment:
```suggestion
        self.table_name = table_name or DEFAULT_TABLE_NAME
```
or better in _init_ use `table_name: str = DEFAULT_TABLE_NAME `.

The same for schema_name

Reply:
Got it.

Reply:
Done.